lib/data/kinetic/Kinetic.js

```diff
@@ -0,0 +1,360 @@
+     */
+    getSockets() {
+        return this._sockets;
+    }
+
+    /**
+     * get the socket at the index.
+     *
+     * @param {number} index - the index of the socket.
+     * @returns {Object} - the socket.
+     */
+    getSocket(index) {
+        return this._sockets[index];
+    }
+
+    /**
+     * get index of the socket.
+     *
+     * @param {string} host - the host of the socket.
+     * @param {string} port - the port of the socket.
+     * @returns {number|object} - the index of the socket or null if it does not
+     *                            exist.
+     */
+    getSocketIndex(host, port) {
+        this._sockets.forEach((socket, index) => {
+            if (socket.host === host && socket.port === port) {
+                return index;
+            }
+            if (index === this._sockets.length - 1) {
+                return null;
+            }
+            return undefined;
+        });
+    }
+
+    _incrString(str) {
+        this._lexiKey = (parseInt(str, 36) + 1).toString(36);
+        return this._lexiKey;
+    }
+
+    /**
+     * send the put request to the right drive.
+     *
+     * @param {number} index - the index of the socket
+     * @param {Buffer} value - the value to be put.
+     * @param {Object} options - the options object for the put request:
+     *                     kinetic-protocol
+     *        {string} options.synchronization - synchronization mode
+     *        {boolean} options.force - option for forcing the put
+     * @param {function} callback - the callback
+     * @returns {undefined}.
+     */
+    put(index, value, options, callback) {
+        this._sockets[index]._request = true;
+        const socket = this._sockets[index];
+        const key = Buffer.concat([Buffer.from(
+            `${socket.drive.host}:${socket.drive.port}:` +
+            `${this._incrString(this._lexiKey)}`)]);
+        const tag = crypto.createHmac('sha1', 'asdfasdf').update(value)
+                  .digest();
+        const pdu = new kineticlib.PutPDU(
+            socket.sequence, socket.connectionID, socket.clusterVersion,
+            key, value.length, tag, options);
+        this.setSequence(index, socket.sequence + 1);
+        const header = pdu.read();
+        const len = header.length + value.length;
+        return socket.sock.write(Buffer.concat([header, value], len), err => {
+            if (err) {
+                return callback(err);
+            }
+            return this._sockets[index]._events.once(
+                'putResponse', err => callback(err, key));
+        });
+    }
+
+    /**
+     * send the get request to the right drive.
+     *
+     * @param {number} index - the index of the socket
+     * @param {Buffer} key - the value to be put.
+     * @param {number[]} range - the range to get
+     * @param {function} callback - the callback
+     * @returns {undefined}.
+     */
+    get(index, key, range, callback) {
```

Comment:
I'm not exactly sure I'm getting this right, but are we putting everything in memory here? It's fine for now as objects are at maximum 1Mo, but we'll circumvent that limit in the future. This warrants at least a `FIXME` or a `TODO` in there, with an associated issue.